### ISSUE TITLE
Add support for foundry

### ISSUE THREAD

**Issue by aldoborrero** [nix-community/ethereum.nix#126]
Adding support for [foundry](https://github.com/foundry-rs/foundry/) would be very nice! Especially if we package it adequately from sources.

I know [`foundry.nix`](https://github.com/shazow/foundry.nix) effort exists; maybe we should talk with @shazow and try to merge it here?

**Comment by shazow** [comment]
Not opposed to having it merged here, but a few caveats/warnings:

First, foundry.nix is packaging from binaries rather than from source. Last time I tried to package from source, there were too many untagged dependencies that it was hard to make it build properly in an immutable way (namely ethers-rs and some other stuff). I haven't checked recently what the state of this is now, I hope it's better now?

Second, there's still no tagged version releases for foundry, only nightly commit binary releases, which are ephemeral (get cleaned up after some number of weeks).

To work around this, foundry.nix does nightly autoupdating to pull the latest nightly release. In addition, we negotiated adding a monthly non-ephemeral nightly release (which unfortunately has been flaky), so we have a `monthly` branch that points to that one.

Not sure if it would be appealing for you to pull in all of these moving pieces. :) But you're very welcome to pull in any subset. I'm happy to generally migrate to this repo if we can achieve somekind of parity here, or keep both running in parallel until we can.

Ideally I was waiting for when there are tagged releases across the dependency set before I was going to try again to repackage.
